pkg/unshare/unshare_linux.go

```diff
@@ -387,10 +388,42 @@ const (
 	UsernsEnvName = "_CONTAINERS_USERNS_CONFIGURED"
 )
 
+// hasFullUsersMappings checks whether the current user namespace has all the IDs mapped.
+// current user namespace
+func hasFullUsersMappings() (bool, error) {
+	content, err := ioutil.ReadFile("/proc/self/uid_map")
+	if err != nil {
+		return false, err
+	}
+	// Both runc and crun check whether the current process is in a user namespace
+	// by looking up 4294967295 in /proc/self/uid_map.  If the mappings would be
+	// copied as they are, the check in the OCI runtimes would fail.  So just split
+	// it in two different ranges.
+	return bytes.Contains(content, []byte("4294967295")), nil
```

Comment:
This comment doesn't seem to describe what's happening before or after it, so I don't understand what's happening here.

Reply:
I've fixed the comment and pushed a new version